packages/api-server/src/base/worker.ts

```diff
@@ -0,0 +1,96 @@
+import * as Sentry from "@sentry/node";
+import { envConfig } from "./env-config";
+import { logger } from "./logger";
+const newrelic = require("newrelic");
+
+const POLL_TIME_INTERVAL = 5000; // 5s
+const LIVENESS_CHECK_INTERVAL = 5000; // 5s
+
+// TODO: use the following class to rewrite BlockEmitter
+export class BaseWorker {
+  protected isRunning: boolean;
+  protected pollTimeInterval: number;
+  protected livenessCheckInterval: number;
+
+  constructor({
+    pollTimeInterval = POLL_TIME_INTERVAL,
+    livenessCheckInterval = LIVENESS_CHECK_INTERVAL,
+  } = {}) {
+    this.isRunning = false;
+    this.pollTimeInterval = pollTimeInterval;
+    this.livenessCheckInterval = livenessCheckInterval;
+  }
+
+  // Main worker
+  async startForever() {
+    await this.start();
+    setInterval(async () => {
+      if (!this.running()) {
+        logger.error(
+          `${this.constructor.name} has stopped, maybe check the log?`
+        );
+        await this.start();
+      }
+    }, this.livenessCheckInterval);
+  }
+
+  async start() {
+    this.isRunning = true;
+    this.scheduleLoop();
+  }
+
+  stop() {
+    this.isRunning = false;
```

Comment:
Should we cancel the [`setInterval`](https://github.com/nervosnetwork/godwoken-web3/pull/514/files#diff-6f4ac3ad134b86291a16807fe74f3e1859eb484139a808df81d38f1d3881e37bR27) task?

Reply:
yeah we should do this

Reply:
second thought: since the setInterval is used in `startForever`, it seems better to keep the interval task which can [alert](https://github.com/nervosnetwork/godwoken-web3/pull/514/files#diff-6f4ac3ad134b86291a16807fe74f3e1859eb484139a808df81d38f1d3881e37bR30) the not-running-forever situation?

Reply:
> second thought: since the `setInterval` is used in `startForever`, it seems better to keep the interval task which can [alert](https://github.com/nervosnetwork/godwoken-web3/pull/514/files#diff-6f4ac3ad134b86291a16807fe74f3e1859eb484139a808df81d38f1d3881e37bR30) the not-running-forever situation?

I'm not sure 😂

Reply:
updated, add new method `stopForever()` to clear the intension to stop and remove interval, and keep `stop()` unchanged.